src/bin/tikv-ctl.rs

```diff
@@ -182,3 +313,113 @@ fn dump_range(db: DB, from: String, to: Option<String>, limit: Option<u64>, cf:
                  })
         .unwrap()
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::str;
+    use rocksdb::Writable;
+    use tikv::util::codec::bytes::encode_bytes;
+    use tikv::util::codec::number::NumberEncoder;
+    use tikv::raftstore::store::keys;
+    use tikv::storage::{ALL_CFS, CF_LOCK, CF_WRITE, CF_DEFAULT};
+    use tikv::storage::mvcc::{Lock, Write, LockType, WriteType};
+    use tempdir::TempDir;
+    use tikv::util::rocksdb::new_engine;
+    use tikv::storage::types::Key;
+
+    const PREFIX: &'static [u8] = b"k";
+
+    #[test]
+    fn test_ctl_mvcc() {
+        let tmp_dir = TempDir::new("mvcc_tmp").expect("create mvcc_tmp dir");
+        let file_path = tmp_dir.path().join("tmp_db");
+        let db = new_engine(file_path.to_str().unwrap(), ALL_CFS).unwrap();
+        let test_data: Vec<(_, _, u64)> =
+            vec![(PREFIX, b"v", 5), (PREFIX, b"x", 10), (PREFIX, b"y", 15)];
+        for &(k, v, ts) in &test_data {
+            let key = keys::data_key(Key::from_raw(&k[..]).append_ts(ts).encoded().as_slice());
+            let value = v;
+            db.put(key.as_slice(), &value[..]).unwrap();
```

Comment:
Just `put(&key, v).unwrap()` should be ok. Don't take a full slice again when the original value is a slice already. If the original value is a `Vec`, use `&v` instead of `&v[..]` if possible.